src/schemas/string/HexColor.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * A valid hexadecimal color value.
+ *
+ * @since 1.0.0
+ */
+
+import * as PB from '../../PatternBuilder'
+import { make } from '../../SchemaExt'
+import { Brand } from 'io-ts'
+import { pipe } from 'fp-ts/function'
+
+/** @internal */
+type HexColorBrand = Brand<{ readonly HexColor: unique symbol }['HexColor']>
+
+/**
+ * A valid hexadecimal color value.
+ *
+ * @since 1.0.0
+ * @category Model
+ */
+export type HexColor = string & HexColorBrand
+
+/**
+ * /^#?([0-9A-F]{3}|[0-9A-F]{4}|[0-9A-F]{6}|[0-9A-F]{8})$/i
+ *
+ * @since 1.0.0
+ * @category Pattern
+ */
+export const hexColor: PB.Pattern = pipe(
+  PB.maybe(PB.char('#')),
+  PB.then(
+    PB.subgroup(
+      pipe(
+        PB.characterClass(false, ['0', '9'], ['A', 'F']),
+        PB.exactly(3),
+        PB.or(pipe(PB.characterClass(false, ['0', '9'], ['A', 'F']), PB.exactly(4))),
+        PB.or(pipe(PB.characterClass(false, ['0', '9'], ['A', 'F']), PB.exactly(6))),
+        PB.or(pipe(PB.characterClass(false, ['0', '9'], ['A', 'F']), PB.exactly(8)))
```

Comment:
`PB.xdigit` (could probably use an alias to `hexDigit`)

also could do something like

```ts
PB.between(3, 4)(PB.xdigit),
PB.or(PB.exactly(6)(PB.xdigit)),
PB.or(PB.exactly(8)(PB.xdigit)),
```

Reply:
not sure if it'll be a common case, but you could also make a pattern combinator like `PB.timesOneOf([3, 4, 6, 8])(PB.xdigit)` which takes the array, maps those values to `PB.exactly`s and chains them together with `or`s

Reply:
Oh, good point!  I forgot about xdigit, I will update that tonight, and the between is nice too 👍